### PR TITLE
[feature fix] Retractions/Embargoes + Archiver

### PR DESCRIPTION
**Pull request by samchrisinger**

### CODE REVIEW

File: website/project/model.py [hunk 1]
```diff
@@ -2398,18 +2398,24 @@ def set_privacy(self, permissions, auth=None, log=True):
...
-                self.registered_from.add_log(
-                    action=NodeLog.EMBARGO_CANCELLED,
-                    params={
-                        'node': self._id,
-                        'embargo_id': self.embargo._id,
-                    },
-                    auth=auth,
-                )
-                self.embargo.save()
-            self.is_public = True
```
Comment (lyndsysimon) [inline website/project/model.py]: This line made nodes public even if they weren't registrations. How are nodes set public after this change?

File: website/project/model.py [hunk 2]
```diff
@@ -2398,18 +2398,24 @@ def set_privacy(self, permissions, auth=None, log=True):
...
+            if self.is_registration:
+                def r_set_public(node):
```
Comment (sloria) [inline website/project/model.py]: What is the purpose of this?

Comment (lyndsysimon) [inline website/project/model.py]: I would rather this not be be a nested function if possible. As it's only two lines, I'd be comfortable with the reptition:

    node.is_public = True
    node.save()
    for child in node.nodes_primary:
        child.is_public = True
        child.save()

If you feel strong that it should be a nested function, please rename it `recursive_set_public`.

File: website/project/model.py [hunk 3]
```diff
@@ -2398,18 +2398,24 @@ def set_privacy(self, permissions, auth=None, log=True):
...
+                    node.is_public = True
+                    node.save()
```
Comment (lyndsysimon) [inline website/project/model.py]: Consider adding a flag `save=True` to this function's parameter list, for consistency with the other methods of `Node`.

File: website/project/model.py (hunk 2 above)
Comment (lyndsysimon) [inline website/project/model.py]: @sloria - if a private registration had children, this iterates them recursively and sets them to public. This situation can be encountered when a legacy private registration is made public.

Comment (sloria) [inline website/project/model.py]: Oh, I didn't see the recursive call on ~L2406. Disregard.

**Comment by lyndsysimon** [comment]
First pass complete.

**Pull request merged**